core/testsupport/CMakeLists.txt

```
@@ -4,6 +4,10 @@
 # higher than kInfo are issued by tests.
 # Stephan Hageboeck, CERN, 2022
 
+if(NOT testsupport)
```

Does it imply/need to have somewhere the equivalent of
```
if (testing)
  set(testsupport ON)
endif()
```
or
```suggestion
if(NOT testsupport AND NOT testing)
```

---

No, because one wants to build the ROOT unit tests `testing=ON`, one also needs the test support library.

---

Yes then I am confused when I see:
```
if(NOT testsupport)
  return()
endif()
```
I assume that this means that the `TestSupport` library will not be build (if `testsupport` is not ON) because the body of the cmake is skipped.  Since the CI test pass, I am indeed missing something.  What am I missing?

---

I was missing the new line 369 of `RootBuildOptions.cmake` which does *_exactly_* my first suggestion.